server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java

```diff
@@ -176,9 +177,13 @@ private CurrentWsResponse toWsResponse(DbSession dbSession, String userLogin) {
     return of(homepage.build());
   }
 
+  private boolean shouldCleanProjectHomepage(Optional<ComponentDto> projectOptional) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Specify a "ComponentDto" parameter instead. ([squid:S3553](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-1185%2FSONAR-10449-add-consistency-validation-on-homepage-retrieval&issues=AWHdOS3gw6XGHKvPuYc4&open=AWHdOS3gw6XGHKvPuYc4)

[ISSUE_KEY]: AWHdOS3gw6XGHKvPuYc4